NumLib/DOF/LocalToGlobalIndexMap.cpp

```diff
@@ -12,6 +12,28 @@
 namespace NumLib
 {
 
+namespace
+{
+
+// Make the cumulative sum of an array
```

Comment:
Minor: Maybe add a comment that the cumulative sums start with zero.
Also minor IMHO: Could be done with std::partial_sum